src/Interpreters/ActionsDAG.cpp

```diff
@@ -229,10 +162,7 @@ ActionsDAG::Node & ActionsDAG::addFunction(
     node.function = node.function_base->prepare(arguments);
 
     /// If all arguments are constants, and function is suitable to be executed in 'prepare' stage - execute function.
-    /// But if we compile expressions compiled version of this function maybe placed in cache,
-    /// so we don't want to unfold non deterministic functions
-    if (all_const && node.function_base->isSuitableForConstantFolding()
-        && (!settings.compile_expressions || node.function_base->isDeterministic()))
```

Comment:
Cannot find, where is this code now?

Reply:
I have understood that we do not need this check anymore.
It is because we calculate hash for compile expressions in a different way.
Before, the whole actions subtree was used as cache key, so expressions like `now() + 1` could not be replaced (so that value of `now() + 1` is different for the next query.
Now, cache key ignore columns names, but uses only function names and constant values. So, if we replace `now() + 1` to constant `1616066911`, cache will only use value `1616066911` itself.